### PR TITLE
 Fix issue creating directories in firefox profiles.

### PR DESCRIPTION
**Pull request by jgraham**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/geckodriver/69)

### CODE REVIEW

File: src/marionette.rs [hunk 1]
```diff
@@ -380,18 +380,35 @@ fn unzip_buffer(buf: &[u8], dest_dir: &Path) -> WebDriverResult<()> {
                 "Processing zip file failed")
         }));
         let unzip_path = {
-            let rel_path = Path::new(file.name());
-            let unzip_path = dest_dir.join(rel_path);
-            if let Some(dir) = unzip_path.parent() {
-                if !dir.exists() {
-                    try!(fs::create_dir_all(dir));
+            let name = file.name();
+            let is_dir = name.ends_with("/");
```
Comment (JohanLorenzo) [inline src/marionette.rs]: What's the advantage of this method compared to [Path.is_dir()](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_dir) ?

Comment (jgraham) [inline src/marionette.rs]: Well the directory doesn't actually exist yet, so `Path.is_dir()` will always be `false`.

**Comment by JohanLorenzo** [comment]
I confirm this patch allows extracting [this archive](https://irccloud.mozilla.com/file/cwxokx9h/xqwdtkc7.blank.zip) correctly. The profile is then correctly pushed to Firefox.

Before the patch, wires just failed with a cryptic error:
```rust
DEBUG:wires::marionette: Got NewSession command
DEBUG:webdriver::command: Getting firefox_profile from capabilities
DEBUG:wires::marionette: Using custom profile
INFO:mozprofile::profile: Using profile path /tmp/rust_mozprofile.TwbAvDzFI2wb
DEBUG:webdriver::server: Returning status InternalServerError
DEBUG:webdriver::server: Returning body {"error":"unknown error","message":"other os error"}
DEBUG:hyper::server::response: writing head: Http11 InternalServerError
DEBUG:hyper::server::response: headers [
Headers { Date: Fri, 03 Jun 2016 10:04:03 GMT, Content-Length: 52, Content-Type: application/json, Connection: close, }]
DEBUG:hyper::server::response: write 52 bytes
DEBUG:hyper::server: keep_alive = false for 127.0.0.1:60516
DEBUG:hyper::server: keep_alive loop ending for 127.0.0.1:60516
ERROR  { [WebDriverError: Unable to parse new session response: {"error":"unknown error","message":"other os error"}] name: 'WebDriverError' }
```

**Comment by andreastt** [comment]
@jgraham Are the changes to Cargo.lock here related to the patch? If not, can you apply that in a separate commit?

**Comment by jgraham** [comment]
Not really related I don't think. I can break this up.

**Comment by andreastt** [comment]
Reviewed 1 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/marionette.rs, line 384 \[r1\]](https://reviewable.io:443/reviews/mozilla/geckodriver/69#-KJaEcwpBSmbffDoDHAY:-KJaEcwqa2jtpQMNLTEH:347258027) ([raw file](https://github.com/mozilla/geckodriver/blob/104ab0ceb0010f45b3646e2959d3ce9747913bf4/src/marionette.rs#L384)):*
> ```Rust
>         let unzip_path = {
>             let name = file.name();
>             let is_dir = name.ends_with("/");
> ```

I’m a bit surprised that [`zip::read::ZipFile`](http://mvdnes.github.io/rust-docs/zip-rs/zip/read/struct.ZipFile.html) does not offer an `is_dir()` function or similar, but this does the trick from what I can tell.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/geckodriver/69)*

**Comment by andreastt** [comment]
:lgtm:

<details><summary><i>Previously, jgraham wrote…</i></summary>
> Not really related I don't think. I can break this up.
</details>


---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/geckodriver/69#-:-KJaFDowjCttKVFaN252:1417088721)*

**Pull request merged**